client/src/services/posts.js

```diff
@@ -17,26 +17,35 @@ export const editPost = (postData, postId) =>
 export const deletePost = postId =>
   axios.delete(`${apiRoute}/delete-post/${postId}`)
 
-export const getPostsByViews = index => {
-  const route = index ? `${apiRoute}/by-views?index=${index}` : `${apiRoute}/by-views`
-  return axios.get(route)
-}
-
-export const getPostsByTag = (tag, index) => {
-  const route = index ? `${apiRoute}/by-tags?tag=${tag}&index=${index}` : `${apiRoute}/by-tags?tag=${tag}`
+export const getPostsByType = (payload) => {
+  let route = `${apiRoute}/get-posts?type=${payload.type}`
+  if (payload.type === 'search') {
+    if (payload.tags) {
+      route = `${route}&tags=${payload.tags}`
+    } else {
+      route = `${route}&emoji=${payload.emoji}`
+    }
+  }
+  if (payload.index && payload.index !== 'Last index') {
+    return axios.get(route, {
+      params: {
+        index: payload.index
+      }
+    })
+  }
   return axios.get(route)
 }
 
-export const getPostsByFollowings = index => {
-  const route = index ? `${apiRoute}/by-followings?index=${index}` : `${apiRoute}/by-followings`
-  return axios.get(route)
+export const getLikedList = (videoId) => {
+  axios.get(`/api/posts/get-post-likes/${videoId}`)
+    .then((response) => {
+      return response.data.result
```

Comment:
When only return is used - try always to simplify things:
then(({data}) => data.result)